### PR TITLE
Bugfix FXIOS-9941 Incorrect behavior when (re)opening the screen

### PR DESCRIPTION
**Pull request by bmihai23**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9941)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21826)

## :bulb: Description
- To address this issue, the solution is to return `true` immediately to ensure proper redirection to the homepage, as described in the bug report
- Additionally, both `PocketViewModelTests` and `StartAtHomeHelperTests`have passed successfully

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift [hunk 1]
```diff
@@ -45,24 +45,20 @@ class StartAtHomeHelper: FeatureFlaggable {
...
-        var timeSinceLastActivity = 0
-        var timeToOpenNewHome = 0
 
-        if setting == .afterFourHours {
-            timeSinceLastActivity = dateComponents.hour ?? 0
-            timeToOpenNewHome = 4
-        } else if setting == .always {
-            timeSinceLastActivity = dateComponents.second ?? 0
-            timeToOpenNewHome = 5
+        switch setting {
```
Comment (FilippoZazzeroni) [inline firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift:54]: LGTM just a nit about code structure. What do you think if we refactor it like this:
```swift
switch startAtHomeSetting {
case .afterFourHours:
  let lastActivityTimestamp = ...
  let dateComponents = ..
  return dateComponents.hour >= 4
case .always:
  return true
case .disabled:
  return false
}
```
I thinks this is more compact but still preserve readability. Let me know what you think

**Review by FilippoZazzeroni — COMMENTED** [review]
Just a nit about code structure. Great work 🎉

**Review by FilippoZazzeroni — APPROVED** [review]
Looks Great @bmihai23 thanks for the contribution 🎉

File: firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift [hunk 2]
```diff
@@ -44,25 +44,19 @@ class StartAtHomeHelper: FeatureFlaggable {
...
-                                                             to: Date())
-        var timeSinceLastActivity = 0
-        var timeToOpenNewHome = 0
-
-        if setting == .afterFourHours {
-            timeSinceLastActivity = dateComponents.hour ?? 0
-            timeToOpenNewHome = 4
-        } else if setting == .always {
-            timeSinceLastActivity = dateComponents.second ?? 0
-            timeToOpenNewHome = 5
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift:62]: please test this behaviour thoroughly. I remember there was a reason why this was done.... I just don't remember what that reason was. lololol

Comment (FilippoZazzeroni) [reply firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift:62]: I didn't know it, thanks @adudenamedruby for the reference. I'll run tests on it 💪

Comment (FilippoZazzeroni) [reply firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift:62]: Ok thanks @bmihai23 that you have restored it, so like this we comply to previous implementation.

File: firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift [hunk 3]
```diff
@@ -44,25 +44,20 @@ class StartAtHomeHelper: FeatureFlaggable {
...
+        switch startAtHomeSetting {
+        case .afterFourHours:
+            let hoursSinceLastActive = dateComponents.hour ?? 0
+            return hoursSinceLastActive >= 4
+        case .always:
+            let secondsSinceLastActive = dateComponents.second ?? 0
+            return secondsSinceLastActive >= 5
```
Comment (FilippoZazzeroni) [inline firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift:57]: Could we add 4 and 5 to a private struct like Constants, so we reference to those number with a constant that has a meaningful name, what do you think ?
```swift
class ... {
private struct Constants {
 ...
 ...
}
}
```

Comment (bmihai23) [reply firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift]: Great suggestion! I've moved the hardcoded values for hours and seconds to a Constants struct to make the code more readable and maintainable

**Review by FilippoZazzeroni — APPROVED** [review]
Great 🎉 🎉

File: firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift (hunk 3 above)
Comment (FilippoZazzeroni) [reply firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift]: Thanks very much @bmihai23 😃

**Pull request merged**